### PR TITLE
Switch password authentication over to SSHJ

### PR DESCRIPTION
**Pull request by FabianHenneke**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring

## :scroll: Description

Switch the SSH backend for SSH password authentication over to SSHJ. Along the way, refactor GitOperation some more and unify the password dialog flows for HTTPS/SSH. As an added bonus, passwords are now no longer kept referenced by static factories.

~~Note that the resource string have some inconsistencies related to the distinction between password/passphrase. I simply preserved those for now, but we should probably resolve them within `GitOperationPasswordFinder` by passing in the `Protocol`.~~

## :bulb: Motivation and Context

One step closer to retiring Jsch.

## :green_heart: How did you test it?
I verified that HTTPS password authentication works as before. I lack the setup to test SSH password authentication though.

## :pencil: Checklist

- [x] I formatted the code with the IDE's reformat action (Ctrl + Shift + L/Cmd + Shift + L)
- [x] I reviewed submitted code

### CODE REVIEW

**Comment by msfjarvis** [comment]
Password auth over SSH works for me.

File: app/src/main/java/com/zeapo/pwdstore/git/GitOperation.kt [hunk 1]
```diff
@@ -16,21 +17,74 @@ import com.google.android.material.textfield.TextInputEditText
...
+import org.eclipse.jgit.transport.URIish
 import java.io.File
+import kotlin.coroutines.Continuation
 import kotlin.coroutines.resume
 
+
+private class GitOperationPasswordFinder(val callingActivity: Activity, val passwordPref: String,
+                                         @StringRes val message: Int) : InteractivePasswordFinder() {
+
+    val encryptedSettings = callingActivity.getEncryptedPrefs("git_operation")
```
Comment (msfjarvis) [inline app/src/main/java/com/zeapo/pwdstore/git/GitOperation.kt:43]: We should probably just inline the `git_operation` string into the method and adjust the name.

File: app/src/main/java/com/zeapo/pwdstore/git/GitOperation.kt [hunk 2]
```diff
@@ -16,21 +17,74 @@ import com.google.android.material.textfield.TextInputEditText
...
+
+    override fun askForPassword(cont: Continuation<String?>, isRetry: Boolean) {
+        val storedPassphrase = encryptedSettings.getString(passwordPref, null)
+        if (isRetry)
+            encryptedSettings.edit { putString(passwordPref, null) }
```
Comment (msfjarvis) [inline app/src/main/java/com/zeapo/pwdstore/git/GitOperation.kt:48]: would `remove` be cheaper?

File: app/src/main/java/com/zeapo/pwdstore/git/GitOperation.kt [hunk 3]
```diff
@@ -77,6 +145,19 @@ abstract class GitOperation(fileDir: File, internal val callingActivity: Activit
         return this
     }
 
+    private fun getSshKey(make: Boolean) {
+        try {
+            // Ask the UserPreference to provide us with the ssh-key
+            // onResult has to be handled by the callingActivity
+            val intent = Intent(callingActivity.applicationContext, UserPreference::class.java)
```
Comment (msfjarvis) [inline app/src/main/java/com/zeapo/pwdstore/git/GitOperation.kt:152]: Some day I will replace this cruft...

**Comment by FabianHenneke** [comment]
I refactored this a bit more and made all UI string depend on the connection mode.

**Pull request merged**